### ISSUE TITLE
Some blending doesn't look good

### ISSUE THREAD

**Issue by MarcelHB** [gemrb/gemrb#2053]
#### Bug description
While fixing and looking at #2050 more closely, I found that our blending doesn't look as good as in the originals, may not the same blend mode? For the character animations it's the `AA_BLEND` flag at play, the spell is probably analogue.

#### Steps to reproduce
1. Cast spells, fire, acid effects ...

#### Expected behavior
See below.

#### Screenshots
GemRB
![blending_gemrb](https://github.com/gemrb/gemrb/assets/238558/d03f48e4-5112-4402-9956-aeed032821dd)

Native
![blending_native](https://github.com/gemrb/gemrb/assets/238558/46a3150d-0bf0-46f8-a9a3-96c94d58f5b4)

GemRB version (check as many as you know apply)
- [x] master as of this issue
- [ ] 0.9.2
- [ ] 0.9.1
- [ ] 0.9.0

Video Driver (check as many as you know apply)
- [ ] SDL1.2
- [ ] SDL2 built with `OPENGL_BACKEND` enabled
- [ ] SDL2 without `OPENGL_BACKEND` enabled

**Comment by bradallred** [comment]
I still question what we are doing with the palettes. Can you add the BAMs in your screenshots to the issue so its easier to investigate?

**Comment by bradallred** [comment]
If it's not a palette issue, then my next guess would be using a different blend mode as you suggest. Projectiles use a two bits to determine which mode to use so we may need to follow that same logic:
https://github.com/gemrb/gemrb/blob/8c6fd7afa02667a3e7514c263361e876ee3e784d/gemrb/core/Projectile.h#L73-L76

**Comment by MarcelHB** [comment]
It's (IWD2) ACIDH.BAM and CONURCG.BAM as shown in NI:
![acid_bams](https://github.com/gemrb/gemrb/assets/238558/bd9c9c16-48c0-49e0-91ec-638ee2aca257)

**Comment by lynxlynxlynx** [comment]
I doubt a bit these two are from projectiles — one is clearly a casting glow (so VVC/BAM in fx_casting_glow).

**Comment by bradallred** [comment]
I know they aren't projectiles, I'm speculating that they ought to behave in a similar fashion in regards to blending. I'm also pointing out that we used to do things incorrectly with projectiles with their palettes/blending and perhaps we were doing the same type of thing here.

In general, I'm skeptical of the type of palette manipulation I removed in 21c16d0 and I doubt it was the only place we were doing it.

> It's (IWD2) ACIDH.BAM and CONURCG.BAM as shown in NI: ![acid_bams](https://private-user-images.githubusercontent.com/238558/310657678-bd9c9c16-48c0-49e0-91ec-638ee2aca257.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3NTkyODYsIm5iZiI6MTcwOTc1ODk4NiwicGF0aCI6Ii8yMzg1NTgvMzEwNjU3Njc4LWJkOWM5YzE2LTQ4YzAtNDllMC05MWVjLTYzOGVlMmFjYTI1Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQyMTAzMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMGRkMzFlYzE1Y2M0MWIxNjdkNWJjNDQ3ODI3Y2UyNzczMDYyOTUxMTg0Yjc0OWUzMTI1NGZhODQ5ZWZmODU1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6pzEDRFYPiv9k8urQC6QZBvhD0-0CPsiEvcxAQ5DuZ8)

since those have a white instead of black background, `ONE_MINUS_DST` probably isn't the correct mode, but maybe one of the others.

**Comment by lynxlynxlynx** [comment]
They're transparent (green), I guess the white is from screenshoting (and the second one is actually CONJUCG.BAM).
For the splash translucency might be enough, but not sure about the casting glow, the brightening could be a visual trick, since the bg is darker.

**Comment by bradallred** [comment]
No, these need special blending. It's the only way to achieve a gradient blend in the original engine and it's clear from the screenshot that there is no gradient. It should be easy to test. If it's a VVC, just force `IE_VVC_BLENDED` to see if that does it. Then if not, go tweak the blending used for the `IE_VVC_BLENDED` case to try the others. It's also possible we need a new blend mode, but I doubt it.

we do have this hack that might need to be tweaked/resolved, but maybe that hack is actually what makes this work with BAMs with a regular color key such as these:
https://github.com/gemrb/gemrb/blob/8b5d2c45901c8685480ff4c0a000dc05bb9ff579/gemrb/plugins/SDLVideo/SDLVideo.cpp#L350-L356

**Comment by lynxlynxlynx** [comment]
That resref is unset in all games I have, so I'd sooner rule the mask as unused. Why reserve space for one if you don't actually provide any files for it? And why would you need it if it was constructed at runtime.

**Comment by bradallred** [comment]
If it were used by anything hardcoded there would still likely be a BAM with the alpha information. Finding it might be a pain since it would be invisible in NI, but I would have suspected a similar name to the BAM its an alpha for.

I don't disagree that it's likely unused, features get axed or have alternatives developed without cleaning up the garbage all the time. But it also wasn't uncommon in that era to embed data files within binaries for speed. Don't we already deal with instances of that?

For now it doesn't matter. The obvious first thing to try is the blending.

I see that `AA_BLEND` does set `IE_VVC_BLENDED`, so then id try changing the blending used. IIRC, `ONE_MINUS_DST` was picked because that was what was being used elsewhere. Look at 1ab8bea for context. What is the `shadowalpha` for? Could that be a problem?

**Comment by bradallred** [comment]
Let's also figure out which `RGBModifier`s are being applied to the palette.

**Comment by lynxlynxlynx** [comment]
It's for actual shadows, a special case like the transparency index before it. Off the top of my head it's important for avatar animations and inventory items.

**Comment by lynxlynxlynx** [comment]
SRC and DST look identical and  get very close for casting glows:
![image](https://github.com/gemrb/gemrb/assets/121515/a07c9db8-240c-48fb-81d0-bcf127786043)

Native from first post:
![blending_native](https://github.com/gemrb/gemrb/assets/238558/46a3150d-0bf0-46f8-a9a3-96c94d58f5b4)
